### PR TITLE
Timothy/config fixes

### PR DESCRIPTION
**Pull request by timhwang21**
Screen Shot 2019-12-01 at 7 17 55 AM

### refactor: don't error on ESlint error in development

Lets us make our rules stricter again so builds fail if we do dumb
stuff.

Also fix existing lint errors and define some reused prop types.

### refactor: remove directory-named-webpack-plugin

1. The behavior granted by this plugin is very unintuitive and increases
   barrier to entry
2. Poor compat with current Webpack / Storybook version

Also add `index.js` for all folders.

### fix(EventCard): Small fixes

- Fix GraphQL Meetup query to give correct values
- Fix date parsing
- Align icons properly
- Run Prettier

### chore(EventCard): Add story

### CODE REVIEW

**Comment by timhwang21** [comment]
@a-trost do you think you can look into this build error when you get a chance? I think I'm configuring the Meetup GQL query wrong, as it keeps trying to fetch `time` instead of `local_time` despite querying for `local_time`. Thus `event.local_time` is undefined causing the error in the build.

File: src/components/shared/eventcard/eventcard.js [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
+  const parsedDate = parse(local_date, 'yyyy-MM-dd', new Date());
+  const parsedDateTime = parse(local_time, 'HH:mm', parsedDate);
   const formattedTime = format(parsedDateTime, 'h:mm a');
 
   return (
     <Article>
       <Box p={32} pb={0} gridArea="date">
-        <Day date={local_date} alignItems={{ _: 'start', sm: 'center' }} />
+        <Day
+          date={parsedDateTime.toISOString()}
```
Comment (codybarr) [inline src/components/shared/eventcard/eventcard.js]: @timhwang21 This should be `date={parsedDate.toISOString()}`.

Comment (timhwang21) [reply src/components/shared/eventcard/eventcard.js]: Shouldn't be a difference here!

**Comment by codybarr** [comment]
> @a-trost do you think you can look into this build error when you get a chance? I think I'm configuring the Meetup GQL query wrong, as it keeps trying to fetch `time` instead of `local_time` despite querying for `local_time`. Thus `event.local_time` is undefined causing the error in the build.

@timhwang21 You just need to change the field in the eventlist component graphql query from `time` to `local_time`:

image

**Comment by timhwang21** [comment]
Ah I see now. There are two duplicate queries -- one in `eventlist` and one in `events`. That was the source of the confusion... I'll remove one.

File: src/components/shared/eventcard/eventcard.js (hunk 1 above)
Comment (codybarr) [reply src/components/shared/eventcard/eventcard.js]: Uh, yep, you're right. Please disregard.

Comment (codybarr) [reply src/components/shared/eventcard/eventcard.js]: Do you need to go through the effort of parsing and converting `local_date`? The meetup API delivers that field to you in ISO 8601 format.

Comment (timhwang21) [reply src/components/shared/eventcard/eventcard.js]: Oh good call out!

**Pull request merged**